python/kserve/pyproject.toml

```
@@ -42,7 +42,7 @@ kubernetes = ">=23.3.0"
 python-dateutil = "^2.8.0"
 numpy = "^1.23.5"
 psutil = "^5.9.0"
-ray = { version = "~2.4.0", extras = ["serve"] }
+ray = { version = "^2.4.0", extras = ["serve"] }
```

should be `^2.6.4` (or `^2.7.0`) once released)

---

poetry's `^2.4.0` is equivalent to pip's `~=2.4` is equivalent to `>=2.4.0,<3`

for maximum compatibility with other libraries that want to require kserve

---

Well, the `test_dataplane.py` fails with at some versions of ray in the range you specify/allow/tolerate -- e.g. I saw failures for Python 3.11 with ray `2.6.1` and `2.6.2`.

So, really, we don't want to allow that range. Or we could do something more bespoke, like ...

- exclude certain versions: https://python-poetry.org/docs/dependency-specification/#inequality-requirements `">= 2.4.0, < 3, !=2.6.1, !=2.6.2"`
- specify versions based on Python version: https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies
  ```Python
  ray = [
    {version = "^2.4.0", python = ">=3.6,<3.11"},
    {version = "^2.6.4", python = ">=3.11"}
  ]
  ```

---

from my side `^2.7.0` is totally fine as well, and indeed the safe choice 👍
```suggestion
ray = { version = "^2.7.0", extras = ["serve"] }
```